### DISCUSSION TITLE
What are the risks of staying on the v3.5.* version of this app?

### DISCUSSION THREAD

**Discussion by wdspider** [gitextensions/gitextensions#11286] [Q&A]
I'm typically a person that enjoys keeping her software up to date running on the latest versions possible; however, I cannot seem to bring myself to update to v4 of GitExtensions due to the loss of dark mode support. The light theme is just way too bright and hard on my eyes.  I understand why the support was removed. This isn't a question about that.

My questions are more around:

1. What am I missing out on by staying on v3.5.4.12724 if I'm perfectly content with how well the app currently works for me?
2. Will the v3.5 branch still receive security updates?
3. Should I continue to hold out hope that dark mode will eventually come back even if it means migrating to a different UX paradigm than WinForms? 
4. Should I actually give up on the app that I've been using since v1 and work on learning a different app that has dark mode support?

**Comment by vbjay** [comment]
You will not get support for git later version updates. Read up on the issues with dark mode. It's not as simple as switching back on. Going from framework to  [.Net](https://dotnet.microsoft.com/en-us/learn/dotnet/what-is-dotnet-framework#:~:text=Some%20key%20differences%20include%3A,accepts%20contributions%20from%20the%20community). The hooking we had used application domains.  As you can see [here](https://learn.microsoft.com/en-us/dotnet/core/porting/#unavailable-technologies) we lost that functionality

 I think griping over loss of dark mode which was awesome and I miss too and the huge amount of keeping current and new features is a silly way to act. As soon as we have a replacement that supports the functionality, I'm sure we will put it back in.

**Reply by wdspider** [reply]
Yes, I'm well aware that hooks within the App Domain is how the WinForms theming was accomplished and the removal of App Domains from .NET is why the theming was ripped out.

Is GitExtensions riding WinForms to the grave where the only hope of bring theme support back is the very unlikely event that App Domains get added back in?
Does the team have a migration plan towards a newer UX framework that has broader theming support?

I'm also not griping over the loss of dark mode either. My eyes are simply physically too light sensitive to use the v4.* version of the tool. I used it for a week to the determine its long term unfeasibility.

I'm merely trying to plan out my next steps particularly since I currently enjoy using this app over others while performing a rough risk assessment for simply staying on v3.5.* until it breaks in hopes that theming support will be added back by then.

**Reply by vbjay** [reply]
Switching UX frameworks is a massive undertaking. There have been talks of a few like Maui or Avolonia. The amount of work to switch is huge. So Winforms it stays till we figure it out. 

Love that you like the product. I totally get the white blindness. I do dark stuff as much as possible. I jumped for joy when Jira finally supported it. I get the tooling I use should make it less straining. But I also don't skip out updates just because it doesn't support dark. I lower screen brightness and adjust contrast. With @RussKie and others actually being on the  inside for Winforms I see greater chances of things like https://github.com/dotnet/winforms/issues/7641 may help us a lot in our application.

**Comment by gerhardol** [comment]
1. You miss out on new features and performance improvements in 4.x. There are adaptions to new bahaviuour in Git like "secure folders" too.
2. GE primarily interacts with the environment with Git, so the risk is relative low. If there are some severe issue updates could be done even if there are no plans right now. Of course, there could be other users that join and maintain a 3.x branch. You can always make your own build.
3. The issue vbjay linked to is targeted to .net9, that hope is not lost.
4. That is up to you to make that decision, is usability or looks more important? It is not like GE is a commercial project, maintainers want happy users. It is of course sad to lose users, but the core team cannot do everything.

The dark mode in GE was i my opinion very nicely done, I miss it a lot, my wife even more (from being in the same room as me).
There are still mono users on *nix that use GE 2.5*
